cmd/skopeo/sync.go

```diff
@@ -647,6 +651,31 @@ func (opts *syncOptions) run(args []string, stdout io.Writer) (retErr error) {
 
 	for _, srcRepo := range srcRepoList {
 		options.SourceCtx = srcRepo.Context
+		// only proceed for top refs if defined
+		if opts.topRefs > 0 {
```

Comment:
Is doing it here the right place at all?

- This would trigger even on explicit single-tag sync configurations in the YAML format. That seems to only result in unnecessary noise.
- I imagine that users might very well want to specify a different limit for different repos. So the implementation should make that possible at least in principle. (Making that _actually_ possible would probably require changing the YAML format, compare https://github.com/containers/skopeo/pull/1792 .)
- (This filtering is enough code that moving it out of already pretty large `main` into a separate function seems worthwhile, either way. That would also allow invoking this code from the two parts of `imagesToCopyFromRegistry`, + `imagesToCopyFromDir`. And writing at least a smoke-test unit test for the filtering logic.)